sycl/doc/extensions/proposed/sycl_ext_intel_dataflow_pipes.asciidoc

```diff
@@ -254,6 +270,26 @@ Type aliases in {cpp} through the `using` mechanism do not change the type of a
   pipe<type_alias, int>::write(0);
 ----
 
+== Host pipe read/write
+
+The read/write member functions of a host pipe have different signatures when they are called from the host side, in which case a `sycl::queue` is added to the parameters.
+
+[source,c++,Host pipe read write members,linenums]
+----
+template <typename name,
+          typename dataT,
+          typename propertiesT = properties<>>
+class pipe {
+  // Blocking
+  static dataT read( const queue &q, memory_order order = memory_order::seq_cst );
+  static void write( const queue &q, const dataT &data, memory_order order = memory_order::seq_cst );
+
+  // Non-blocking
+  static dataT read( const queue &q, bool &success_code, memory_order order = memory_order::seq_cst );
+  static void write( const queue &q, const dataT &data, bool &success_code, memory_order order = memory_order::seq_cst );
+}
+----
+
```

Comment:
It will be nice to provide some example of calling host pipe if it is different from the normal pipes.

The example demonstrates where this call resides, input etc, In this case, I think the following will suffice?

```cpp
int data = 3;
using pipe_prop = decltype(experimental::properties{min_capacity<5>})
using my_pipe = pipe<class some_pipe, int>;
myQueue.submit([&](handler& cgh) {
    // enqueue kernels
});
my_pipe::write( myQueue, data);

myQueue.submit([&](handler& cgh) {
    // enqueue a kernel that uses data previously written from host
    int data = my_pipe::read();
});
```

Reply:
@pcolberg Can close this one too.